cli/src/pcluster/templates/cdk_builder.py

```diff
@@ -12,15 +12,151 @@
+from aws_cdk import aws_ssm as ssm
 from aws_cdk import core
 
 from common.utils import load_yaml
 from pcluster.cluster import Cluster, Fsx, HeadNode
+from pcluster.imagebuilder import ImageBuilder
+
+
+class ImageBuilderStack(core.Stack):
+    """Create the Stack for imagebuilder."""
+
+    def __init__(self, scope: core.Construct, construct_id: str, imagebuild: ImageBuilder, **kwargs) -> None:
+        super().__init__(scope, construct_id, **kwargs)
+
+        # TODO: use attributes from imagebuilder config instead of using these static variables.
+        # enable_nvidia = "false"/"true"
+        core.CfnParameter(self, "EnableNvidia", type="String", default="false", description="EnableNvidia")
+        # enable_dcv = "false"/"true"
+        core.CfnParameter(self, "EnableDCV", type="String", default="false", description="EnableDCV")
+
+        # TODO: get archecture by ami id
+        archecture = "x86_64"
+
+        # Setup ImageBuilder Resources
+        # InstanceRole
+        # TODO: use imagebuild.config.build_config.instance_role to create custom one
+        instancerole = iam.CfnRole(
+            self,
+            id="InstanceRole",
+            managed_policy_arns=[
+                core.Fn.sub("arn:${AWS::Partition}:iam::aws:policy/AmazonSSMManagedInstanceCore"),
+                core.Fn.sub("arn:${AWS::Partition}:iam::aws:policy/EC2InstanceProfileForImageBuilder"),
+            ],
+            assume_role_policy_document={
+                "Statement": {
+                    "Action": "sts:AssumeRole",
+                    "Effect": "Allow",
+                    "Principal": {"Service": "ec2.amazonaws.com"},
+                },
+                "Version": "2012-10-17",
+            },
+            path="/executionServiceEC2Role/",
+        )
+        instancerole.add_metadata("Comment", "Role to be used by instance during image build.")
+
+        # InstanceProfile
+        iam.CfnInstanceProfile(
+            self, id="InstanceProfile", path="/executionServiceEC2Role/", roles=[core.Fn.ref("InstanceRole")]
+        )
+
+        # TODO add S3 bucket
+
+        # InfrastructureConfiguration
+        imagebuilder.CfnInfrastructureConfiguration(
+            self,
+            id="PClusterImageInfrastructureConfiguration",
+            name="PCluster-Image-Infrastructure-Configuration",
+            instance_profile_name=core.Fn.ref("InstanceProfile"),
+            terminate_instance_on_failure=imagebuild.config.build_config.terminate_on_failure,
+            instance_types=["p2.xlarge", "c5.xlarge"] if archecture == "x86_64" else ["m6g.xlarge"],
+            key_pair=imagebuild.config.build_config.key_pair_name,
+        )
+
+        # Component
+        if imagebuild.config.dev_settings_config and imagebuild.config.dev_settings_config.update_and_reboot:
+            imagebuilder.CfnComponent(
+                self,
+                id="PClusterComponent",
+                name="PCluster",
+                version="0.0.1",
```

Comment:
Are these attributes required?
If so let's use a constant for the version and use it  everywhere